gobblin-service/src/main/java/org/apache/gobblin/service/modules/core/D2Announcer.java

```diff
@@ -0,0 +1,17 @@
+package org.apache.gobblin.service.modules.core;
+
+
+/**
+ * Interface for marking up/down D2 servers on gobblin service startup. This is only required if using delayed announcement.
+ */
+public interface D2Announcer {
+  /**
+   * Mark up D2 servers
+   */
+  void markUpServers();
```

Comment:
Is it marking all servers as "up" or just the current one?

Reply:
The API from D2 we will use gives a server list, even though I think there will just be one server per host in this case.

Reply:
I suggest to double-check with D2 docs/code that it will actually only mark the current server. If it is going to mark all servers as down, that wouldn't help us with routing data to master node.

If it indeed just works with the current server, then it looks like a confusing API on D2 side. We don't have to carry it inside our code base, and can make it a bit more correct and meaningful in our context.

Reply:
Well if I understand correctly it would be the D2 servers associated with the current host, I don't think it would be possible to markup/markdown D2 servers for other hosts.

But sure, since I think it should just be one server in our case I can change it.